### PR TITLE
Add dg extra sections

### PR DESCRIPTION
**Pull request by theodorekwok**
links to #82 
https://github.com/theodorekwok/tp/blob/theodorekwok-dg-extra-sections/docs/DeveloperGuide.md

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -365,8 +387,171 @@ appropriate message through the `TextUi` class.
 
 ## Glossary
 
-* *glossary item* - Definition
+* *Instrument* - Represents assets that can be traded. Most common examples are stocks and foreign currency.
+* *Etf* - Known as Exchange Traded Funds, they are a type of instrument that tracks the performance of a particular asset. 
+* *Forex* - Foreign exchange market for trading currencies. An example include is the USDSGD exchange rate.
```
Comment (williamlamjy) [inline docs/DeveloperGuide.md:392]: included*

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -365,8 +387,171 @@ appropriate message through the `TextUi` class.
...
+Feel free to come up with more test cases to try for yourself.
+
+**Launch and start up**
+
+1) Ensure that you have `Java 11` installed.
+2) Download the latest jar file [here](https://github.com/AY2122S1-CS2113T-T12-1/tp/releases). 
+3) In your terminal under the directory where the jar file is saved type `java -jar [CS2113T-T12-1][mTracker].jar`.
+   1) If it is successful you should see a mTracker greet message. If you get an error message please create a new issue
+      [here](https://github.com/AY2122S1-CS2113T-T12-1/tp/issues) along with a description of the error.
+4. Refer to the [userguide](https://github.com/AY2122S1-CS2113T-T12-1/tp/blob/master/docs/UserGuide.md) to understand how
```
Comment (williamlamjy) [inline docs/DeveloperGuide.md:408]: 4)* idk why got one random spacing might be this formatting

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -365,8 +387,171 @@ appropriate message through the `TextUi` class.
...
+mTracker$main> done 7
+```
+Expected: An error message that says instrument is already done.
+```
+Instrument at provided index has already been marked as completed!
+```
+
+**Find functionality Testing**
+
+To test the done functionality, there are a few test cases you can try:
```
Comment (williamlamjy) [inline docs/DeveloperGuide.md:524]: find*

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -365,8 +387,171 @@ appropriate message through the `TextUi` class.
...
+mTracker$main> view 8 10
+```
+Expected: It should return only the 8th instrument in the list ignoring the value `10`.
+
+**Loading storage file testing**
+
+To test the program against corruption of saved file data, there are a few test cases you can try:
+1. Testcase: In the saved file on a newline write `This is a fake instrument`.
+
+   Expected: It should say that incorrect instrument type is provided and that instrument would be ignored.
```
Comment (williamlamjy) [inline docs/DeveloperGuide.md:557]: add expected output for this?

**Review by williamlamjy — COMMENTED** [review]
first glance review might add more later. just some typos but good work

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -328,12 +346,16 @@ appropriate message through the `TextUi` class.
...
-
-{Describe the target user profile}
+ * Busy individuals that need a convenient way to record financial information on the go.
+ * Familiar with using the terminal and command-line interface applications.
+ * Individuals that consistently keep up to date with financial news and events.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+Given that financial information is rapidly evolving and growing beyond what the standard brokerages
```
Comment (williamlamjy) [inline docs/DeveloperGuide.md:355]: Remove "Given that"

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -328,12 +346,16 @@ appropriate message through the `TextUi` class.
...
+and traditional financial news provide. It is readily and easily accessible to any individual with an internet connection
```
Comment (williamlamjy) [inline docs/DeveloperGuide.md:356]: This information is readily and easily.....

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -365,8 +387,175 @@ appropriate message through the `TextUi` class.
 
 ## Glossary
 
-* *glossary item* - Definition
+* *Instrument* - Represents assets that can be traded. Most common examples are stocks and foreign currency.
+* *Etf* - Known as Exchange Traded Funds, they are a type of instrument that tracks the performance of a particular asset. 
+* *Forex* - Foreign exchange market for trading currencies. An example included is the USDSGD exchange rate.
+* *Crypto* - Cryptocurrencies are digital currencies that is secured by cryptography methods.
```
Comment (williamlamjy) [inline docs/DeveloperGuide.md:393]: remove "Cryptocurrencies are"

**Review by williamlamjy — COMMENTED** [review]
some rephrasing edits

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (KVignesh122) [reply docs/DeveloperGuide.md]: Rephrase to "Digital currencies that are secured by cryptography methods.

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -390,7 +390,7 @@ such information in a quick and easy way.
 * *Instrument* - Represents assets that can be traded. Most common examples are stocks and foreign currency.
 * *Etf* - Known as Exchange Traded Funds, they are a type of instrument that tracks the performance of a particular asset. 
 * *Forex* - Foreign exchange market for trading currencies. An example included is the USDSGD exchange rate.
-* *Crypto* - Cryptocurrencies are digital currencies that is secured by cryptography methods.
+* *Crypto* - Digital currencies that is secured by cryptography methods.
```
Comment (KVignesh122) [inline docs/DeveloperGuide.md:393]: Digital currencies that are secured by cryptography methods.

**Review by KVignesh122 — COMMENTED** [review]
Just small grammar mistake. Are should be used instead of is.

**Pull request merged**